src/base/bittorrent/session.cpp

```diff
@@ -296,6 +317,8 @@ Session::Session(QObject *parent)
     , m_encryption(BITTORRENT_SESSION_KEY("Encryption"), 0)
     , m_isForceProxyEnabled(BITTORRENT_SESSION_KEY("ForceProxy"), true)
     , m_isProxyPeerConnectionsEnabled(BITTORRENT_SESSION_KEY("ProxyPeerConnections"), false)
+    , m_chokingAlgorithm(BITTORRENT_SESSION_KEY("ChokingAlgorithm"), 0, clampValue(0, 2))
```

Comment:
>libtorrent states its incomplete & untested.
Also read Q: Won’t BitTyrant hurt overall BitTorrent performance if everyone uses it?
in http://bittyrant.cs.washington.edu/

+1.
Don't forget to rebase.